Tasks/Gradle/sonarqube-common/sonarqube-qualitygate.ts

```diff
@@ -0,0 +1,42 @@
+/// <reference path="../../../definitions/vsts-task-lib.d.ts" />
+
+import tl = require('../testlib/tasklib-wrapper');
+
+// Creates a build summary section to display the quality gate status.
+// Returns null if analysisStatus was null.
+export function createBuildSummaryQualityGateSection(analysisStatus:string):string {
```

Comment:
As mentioned above, it would be better to have a ReportBuilder class that takes care of the creating the report file. It can either call sqRest directly to fetch the quality gate status, or it can be given the status (in which case it would be less bound). Either way, there should be a single class with one or maybe two public methods.